### PR TITLE
[HierarchicalListExample]: Custom hierarchical list example.

### PR DESCRIPTION
**Pull request by Kuznietsov**
#### Issue: #609, #1438

A custom `Hierarchical List` example was added.

https://github.com/user-attachments/assets/7fe1a8c8-7f71-41bf-9c36-0d8af66155ad

### CODE REVIEW

File: app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                                    rawProps={ {
+                                        'aria-label': row.isFolded ? 'Unfold' : 'Fold',
+                                        role: 'button',
+                                    } }
+                                    key="icon"
+                                    icon={ FoldingArrow }
+                                    cx={ [
+                                        uuiElement.foldingArrow, uuiMarkers.clickable,
+                                    ] }
+                                    style={ { fill: 'var(--uui-icon)' } }
```
Comment (AlekseyManetov) [inline app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx:145]: let's move it to the css class

File: app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx
Comment (AlekseyManetov) [inline app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx:1]: Let's move inline styles into css classes to make example more pure and readable

File: app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        } else {
+            content = (
+                <Text fontWeight="600" fontSize="16">
+                    {row.value.name}
+                </Text>
+            );
+        }
+        
+        const getIndent = () => {
+            return 12 + (row.indent - 1) * 24;
```
Comment (AlekseyManetov) [inline app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx:116]: If 12 it's a global padding for all items, let's set it via css and exclude from this calculation

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx (hunk 1 above)
Comment (Kuznietsov) [reply app/src/docs/_examples/dataSources/CustomHierarchicalList.example.tsx:145]: Done.

**Comment by NatalliaAlieva** [comment]
Released in 5.9.2 ver.

**Pull request merged**